dvc/command/stages.py

```diff
@@ -0,0 +1,312 @@
+            return prepare_stage_name(up, link=False, use_name=use_name)
+
+        fragments = []
+        for idx, ancestor in enumerate(ancestors):
+            if ancestor is stage:
+                continue
+            if idx > 0:
+                fragments.append(", ")
+            fragments.append(fork)
+            fragments.append(gen_name(ancestor, stage))
+
+        if fragments:
+            text = Text()
+            for fragment in fragments:
+                text.append(fragment)
+            text.truncate(MAX_TEXT_LENGTH, overflow="ellipsis")
+
+    return text
+
+
+def prepare_stage_name(stage: "Stage", link=False, use_name=False) -> Text:
+    title = Text(overflow="fold")
+    address = stage.addressing
+
+    linked = Style(link="file://" + os.path.abspath(stage.relpath))
+    file_style = Style(color=DVC_BLUE, bold=True, underline=False)
+    name_style = None
+    if link:
+        name_style = linked
+        file_style += linked
+
+    if isinstance(stage, PipelineStage):
+        from_pwd = stage.name == address
+        if from_pwd or use_name:
+            title.append(stage.name, style=name_style)
+        else:
+            title.append(stage.relpath, style=file_style)
+            title.append(f":{stage.name}")
+    else:
+        title.append(address, style=file_style)
+    return title
+
+
+def prepare_description(stage: Stage) -> Union[Markdown, Text]:
+    # Description
+    if stage.desc:
+        # not sure if this was ever intended, but markdown looks nice
+        return Markdown(stage.desc)
+
+    description = generate_description(stage)
+    text = Text(description.strip(), style="blue")
+    text.truncate(MAX_TEXT_LENGTH, overflow="ellipsis")
+    return text
+
+
+def list_layout(stages: Iterable[Stage], graph: "nx.DiGraph" = None) -> None:
+    """ Displays stages in list layout using rich """
+
+    LAYOUT_WIDTH = 80
+    LEFT_PAD = (0, 0, 0, 4)
+
+    @render_group()
+    def render_stage(stage: "Stage", idx: int):
+        """Yields renderables for a single repo."""
+        if idx:
+            # separator
+            yield Rule(style=DVC_LIGHT_ORANGE)
+            yield ""
+
+        # title
+        title_table = Table.grid(padding=(0, 1), expand=True)
+        title = prepare_stage_name(stage, link=True)
+
+        # basic info at the right side of the table
+        info_text = Text()
+        info = get_info(stage)
+        for fragment in prepare_info(info):
+            info_text.append(fragment)
+        title_table.add_row(title, info_text)
+
+        # pushing all columns except the first one with title, to the right
+        for idx, column in enumerate(title_table.columns):
+            if not idx:
+                continue
+            column.no_wrap = True
+            column.justify = "right"
+        yield title_table
+        yield ""
+
+        # Table of ancestor nodes
+        nodes_table = Table.grid(padding=(0, 1), expand=True)
```

Comment:
Not sure if we need to so DAG, or just the next-to nodes. We could change it later.

Reply:
Talked with Dmitry on this. He said that the dependent nodes are a good default choice. We will come to it later if needed.